tools/html_transformer/html_transformer.js

```diff
@@ -0,0 +1,95 @@
+var Promise = require('es6-promise').Promise;
+var jsdom = require('jsdom');
+
+exports.window = null;
+
+exports.load = function(filename) {
+  return new Promise(function(resolve, reject) {
+    jsdom.env(filename, function(errors, window) {
+      if (errors && errors.length) {
+        return reject(errors[0]);
+      }
+
+      exports.window = window;
+      return resolve();
+    });
+  });
+};
+
+exports.getSharedDependencies = function() {
+  var doc = exports.window.document;
+  var links = getSharedElements(doc, 'link').map(function(link) {
+    return link.href;
+  });
+  var scripts = getSharedElements(doc, 'script').map(function(script) {
+    return script.src;
+  });
+  var commented = flatmap(getComments(doc), getSharedElementsFromComment);
+  return links.concat(scripts).concat(commented);
+};
+
+function getSharedElements(doc, selector) {
+  return querySelectorAll(doc, selector).filter(isShared);
+}
+
+function getSharedElementsFromComment(comment) {
+  var sharedLink = /<link\s.*href="(\/?shared\/.+)".*>/;
+  var sharedScript = /<script\s.*src="(\/?shared\/.+)".*>/;
+  var links = comment.match(sharedLink) || [];
+  var scripts = comment.match(sharedScript) || [];
+  var results = links.concat(scripts);
+  return results.filter(function(result) {
+    // RegExp.prototype.match picks up the whole thing as well as things
+    // in parens. We only want the urls.
+    return result.indexOf('<') === -1;
+  });;
+};
+
+// Neither document.createTreeWalker nor document.createNodeIterator
+// are supported by jsdom so we're going to do this the old fashioned way...
+function getComments(doc) {
+  var results = [];
+  var queue = [doc.body];
+  while (queue.length) {
+    var next = queue.shift();
+    if (next.nodeType === 8 /* COMMENT_NODE */) {
```

Comment:
But then I'd have to pass a window into this thing just for that so I might just leave this number magical